### PR TITLE
Upgrade apache dubbo version 

### LINKED ISSUE #5523: [Task] <Upgrade apache dubbo version support>
### Description

Now we support apache dubbo 2.x version. We need to upgrade the dubbo version 3.x to meet java17 compatible.  

now

```
<apache.dubbo.version>2.7.15</apache.dubbo.version>
```

please make sure this can running with java17 jdk

### Task List

- [ ] Upgrade apache dubbo version support
- [ ] Upgrad and test pass in shenyu-examples/shenyu-examples-dubbo shenyu-examples/shenyu-examples-sdk 
- [ ] Upgrad and test pass in shenyu-integrated-test/shenyu-integrated-test-apache-dubbo  shenyu-integrated-test/shenyu-integrated-test-k8s-ingress-apache-dubbo

### PR DESCRIPTION
**Pull request by Aias00**
Upgrade apache dubbo version support Fixes #5523

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by Aias00** [comment]
@loongs-zhang workflows are all passed, if u have time , pls help review this 😊

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service/pom.xml [hunk 1]
```diff
@@ -113,6 +119,11 @@
             <artifactId>curator-recipes</artifactId>
             <version>${curator.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.curator</groupId>
+            <artifactId>curator-x-discovery</artifactId>
+            <version>5.1.0</version>
```
Comment (loongs-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service/pom.xml:125]: Why not use `${curator.version}`?

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-xml/pom.xml [hunk 2]
```diff
@@ -112,6 +112,11 @@
             <artifactId>curator-recipes</artifactId>
             <version>${curator.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.curator</groupId>
+            <artifactId>curator-x-discovery</artifactId>
+            <version>5.1.0</version>
```
[the remark this answers is not in the document]
Comment (Aias00) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-xml/pom.xml]: have changed

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service/pom.xml (hunk 1 above)
Comment (Aias00) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service/pom.xml]: changed

**Review by loongs-zhang — APPROVED** [review]

**Pull request merged**